### PR TITLE
Replace tendermint with lazyledger-core

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR manually replaces the imports from "github.com/tendermint/tendermint" to "github.com/lazyledger/lazyledger-core" in all go and proto files. It also changes the methods of `baseapp.BaseApp` to adhere to the new lazyledger-core `ABCI` interface and a few other minor changes to address compile time errors. It should be noted that the new method is not implemented, or well documented and simply passes all transaction data. This PR is required for before #1 can be finished.

closes: #2 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [X] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [X] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
this PR only has a template test for the new method added, as the method is not currently implemented.
- [X] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [X] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [X] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -46,11 +47,10 @@ require (
 	github.com/tendermint/btcd v0.1.1
 	github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15
 	github.com/tendermint/go-amino v0.16.0
-	github.com/tendermint/tendermint v0.34.1
 	github.com/tendermint/tm-db v0.6.3
 	golang.org/x/crypto v0.0.0-20201221181555-eec23a3978ad
 	golang.org/x/net v0.0.0-20200930145003-4acb6c075d10 // indirect
-	google.golang.org/genproto v0.0.0-20201214200347-8c77b98c765d
+	google.golang.org/genproto v0.0.0-20210114201628-6edceaf6022f
```
Comment (liamsi) [inline go.mod:53]: Can this be related to the genproto failures?

Comment (evan-forbes) [reply go.mod:53]: It could be. Everything started working for me after running `go clean -modcache -i -r` :grimacing:

**Comment by liamsi** [comment]
Is this ready for another review?

**Comment by evan-forbes** [comment]
I was just about to comment, I think so. First failing check, Lint/golangci-lint has to do with file permissions that have not been changed, and are present in the cosmos-sdk, but I can change them if anyone thinks it's a good idea. If not, should this check be kept?

The second is failing because the check is comparing the master branch of the sdk with ours, and the sdk has added the endpoints since we last updated. I can update also update to the latest master branch of the sdk if that is desired. If not, should this check be kept? Altered to check our fork?

**Comment by liamsi** [comment]
Thanks! I'll review more thoroughly this week! 

Regarding the `Error: G306: Expect WriteFile permissions to be 0600 or less (gosec)`: which files is it complaining about (I don't see it in the CI output). 

The second test you mention is the proto_breakage? No need to update to the lastest master yet

**Comment by evan-forbes** [comment]
There are quite a few files afaict. Most are in testing, so I don't think those pose that great a threat. There is [this one](https://github.com/lazyledger/cosmos-sdk/blob/73d1760612a2877f0a51dd7ff3e1286b2ec3ae3b/server/config/toml.go#L198), but I used the same permissions as the [cosmos-sdk version](https://github.com/cosmos/cosmos-sdk/blob/d4e63ffe60b63c466dce741624353374f2916144/server/config/toml.go#L209).

File: go.mod [hunk 2]
```diff
@@ -26,6 +26,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.2
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
 	github.com/hashicorp/golang-lru v0.5.4
+	github.com/lazyledger/lazyledger-core v0.0.0-20210115223437-eff282ad2592
```
Comment (liamsi) [inline go.mod:29]: One general remark on the module (the cosmos-sdk one): I was wondering if we should rename it from
`module github.com/cosmos/cosmos-sdk` to `module github.com/lazyledger/cosmos-sdk`. 
I'm almost surprised to see that the replace directive worked in the ll-app without a problem in this case. 

Let's update the app when this is merged and see if this works without any further changes.

**Review by liamsi — APPROVED** [review]
LGTM 💪🏼

**Comment by liamsi** [comment]
Regarding gosec. IMO, it complains in this branch but not in the SDK because, here it recognizes the pattern ioutil.WriteFile while it doesn't with tmos.WriteFile:

 -	err = tmos.WriteFile(file, contents, 0644)
 +	err = ioutil.WriteFile(file, contents, 0644)

This can be silenced by sticking to that `tmos` import instead. Still weird, especially that 0777 place you mentioned above.

**Pull request merged**